# Group Increments
Let us try some calculations in group increments using python as our calculator. First we must load in a data set. I will take the file from the moodle site and place it in the same directory as this notebook. This data was taken from table 2.4 (pg 81) of the textbook.

The following code will import the data.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats

data_file = "Group_Increments.csv"    # This is the data file in the GitHub repository

############################
### Set global variables ###
############################

data_path = "https://raw.githubusercontent.com/blinkletter/4410PythonNotebooks/main/Class_10/data/"
image_path = "https://raw.githubusercontent.com/blinkletter/4410PythonNotebooks/main/Class_10/images/"

github_data_location = "https://raw.githubusercontent.com/blinkletter/LFER-QSAR/main/data/"
github_styles_location = "https://raw.githubusercontent.com/blinkletter/LFER-QSAR/main/styles/"



data_file = github_data_location + data_file

df1 = pd.read_csv(data_file,    # create pandas dataframe from data file
                  delimiter = ",",           # data is separated by commas
                  index_col="group",         # use the column with "group" as a header for the index
                  comment = "#",             # ignore lines in the file that are comments
                  skipinitialspace = True    # ignore extra spaces
                 ) 

#display(df1.head())     # display first five rows at head of Pandas dataframe
display(df1)

dHf
group           
C-H3C     -10.20
C-H2C2     -4.93
C-HC3      -1.90
C-C4        0.50
Cd-H2       6.26
...          ...
N-(CO)C2   14.40
N-(CO)CbH   0.40
N-(CO)2H  -18.50
N-(CO)2C   -5.90
N-(CO)2Cb  -0.50

[85 rows x 1 columns]

## Manipulating the Data
the data tablein in units of kcal/mole. We usually use kJ/mole so let us convert the numbers to that unit and add a new column.

In [2]:
n = 5
pd.set_option('display.max_rows', n) #replace n with the number of rows you want to see completely

df1["dHfkJ"] = df1["dHf"]*4.18        # add a series in units of Kj/mole
display(df1)

dHf    dHfkJ
group                    
C-H3C     -10.20 -42.6360
C-H2C2     -4.93 -20.6074
...          ...      ...
N-(CO)2C   -5.90 -24.6620
N-(CO)2Cb  -0.50  -2.0900

[85 rows x 2 columns]

## Using the Data
We will extract a single data series out of the dataframe. It will still be indexed by "group". let us grab just the series that is in kJ/mole and then we can access individual values using the index.

In [3]:
i=df1["dHfkJ"]             # extract a series of group increments in kJ/mole as "i"
display(i.head())

group
C-H3C    -42.6360
C-H2C2   -20.6074
C-HC3     -7.9420
C-C4       2.0900
Cd-H2     26.1668
Name: dHfkJ, dtype: float64

## Example Calculation
Let us demonstrate the usefulness of using a notebook as a calculator in thermochemistry. Our calculation will be fully documented in the code and any error and be identified and corrected. Others can see exactly how the result was obtained and provide critical analysis of our work. Scientists love to be criticised. (Really, we do. That means somebody read our paper!)

We can answer one of the problems in 8 from chapter 2. What is the calculated heat of formation of adamantane (and compare it to the experimental value).

![adamantane](https://raw.githubusercontent.com/blinkletter/4410PythonNotebooks/main/Class_05/images/adamantane.png)


In [4]:
CH2 = i["C-H2C2"]       # increment for CH2 groups
CH = i["C-HC3"]         # increment for CH groups

print(f"The group increment for C-H2C2 is {CH2:+.2f}")
print(f"The group increment for C-HC3 is {CH:+.2f}")

Hf = 6*CH2 + 4*CH

print(f"the calculated delta Hf is {Hf:+.2f} kJ/mole")

The group increment for C-H2C2 is -20.61
The group increment for C-HC3 is -7.94
the calculated delta Hf is -155.41 kJ/mole


We have calculate the heat of formation of adamantane using group increments as shown above.  The reported observed heat of formation is indicated in problem 8 of chapter 2 and is reported to be -134.25 kJ/mole (-32.12 kcal/mole). We will convert to kJ/mole and then calculate the difference in energy. (Note that the NIST website quotes values from -129 to -139 kJ/mole) 

In [5]:
ObsE = -32.12*4.184
print(f"the observed delta Hf is {ObsE:+.2f} kJ/mole")
print(f"the calculated delta Hf is {Hf:+.2f} kJ/mole")

print(f"the strain energy is therefore {ObsE-Hf:+.2f} kJ/mole")

the observed delta Hf is -134.39 kJ/mole
the calculated delta Hf is -155.41 kJ/mole
the strain energy is therefore +21.02 kJ/mole


Below is another way to do the same math. here I am using the entries in the table directly and splitting the lines (using the "\\") so that each entry is on its own line.

In [6]:
Hf2 = \
   + 6*i["C-H2C2"]  \
   + 4*i["C-HC3"]

print(f"the calculated delta Hf is {Hf2:+.2f} kJ/mole")

the calculated delta Hf is -155.41 kJ/mole


### A note on the data
I didn't notice it being clearly stated in the textbook but all data is from gas-phase condition in the thermochemistry table (table 2.4). The data from the NIST webbook gives the $\Delta H_f$ to be between $-129$ and $-139\; kJ/mole$

The most accepted value of $-134.4\;kJ/mole$ comes from J.B. Pedley, R.D. Naylor, S.P. Kirby, "Thermochemical Data of Organic Compounds", *Chapman and Hall, New York*, **1986**, 1-792

## Another Example

What are the strain energies in cyclobutanone vs. cyclobutanol. Compare with strain energies in cyclohexanone and cyclohexanol? Which do you think will favour reduction?

According to the NIST webbook here are some observed experimental values...

| Name          |     ∆Hf˚ (kJ/mole)|
| :------       | :------           |
| Cyclobutanone | -91.6 or -101.1   |
| Cyclobutanol  |  -145             |
| Hexanone      | -231.1 or -226    |
| Hexanol       |   -290.8          |
| Methylenecyclobutane   | -106 or -121.5 |
| Norbornane    |   -54.9           |

And the observed heats of hydrogenation from NIST...

| Name               |     ∆Hr˚ (kJ/mole)|
| :------            | :------           |
| Cyclobutanone + H2 | -68 or -53        |
| Hexanone + H2      | -75.9 or -63.5    |
| 2-butanone +H2     |  -54.2            |
| 3-pentanone + H2   |  -56.7            |
| 2-pentanone + H2   |  -54.7            |
| propanone + H2     | -55.23            |





In [7]:
# Hf for Cyclobutanone

Hf = \
   + 1*i["C-H2C2"]  \
   + 2*i["C-(CO)CH2"]  \
   + 1*i["CO-C2"]

print(f"Cyclobutanone: Hf is {Hf:+.2f} kJ/mole.  Strain energy is {-91.6 - Hf:+.2f} kJ/mole")

# Hf for Cyclobutanol

Hf = \
   + 3*i["C-H2C2"]  \
   + 1*i["C-OC2H"]  \
   + 1*i["O-CH"]

print(f"Cyclobutanol: Hf is {Hf:+.2f} kJ/mole.  Strain energy is {-145 - Hf:+.2f} kJ/mole")

# Hf for Cyclohexanone

Hf = \
   + 5*i["C-H2C2"]  \
   + 2*i["C-(CO)CH2"]  \
   + 1*i["CO-C2"]

print(f"Cyclohexanone: Hf is {Hf:+.2f} kJ/mole.  Strain energy is {-231.1 - Hf:+.2f} kJ/mole")

# Hf for Cyclohexanol

Hf = \
   + 5*i["C-H2C2"]  \
   + 1*i["C-OC2H"]   \
   + 1*i["O-CH"]

print(f"Cyclohexanol: Hf is {Hf:+.2f} kJ/mole.  Strain energy is {-290.8 - Hf:+.2f} kJ/mole")

# Hf for Methylenecyclobutane  ∆Hf = 121.5 (NIST)

Hf = \
   + 1*i["C-H2C2"]  \
   + 2*i["C-CdCH2"]   \
   + 1*i["Cd-C2"]   \
   + 1*i["Cd-H2"]

print(f"\nMethylenecyclobutane: Hf is {Hf:.2f} kJ/mole.  Strain energy is {121.5 - Hf:+.2f} kJ/mole")


# print(f"\nTest compared to textbook - Calculated methylenecyclobutene: Hf is {Hf:.2f} kJ/mole. \
#         \nStrain energy is {(121.5 - Hf):+.2f} kJ/mole")
print(f"Compare to value of {29*4.183:.2f} given in figure 2.15 of book")

# Hf for Norbornane  ∆Hf = 121.5 (NIST)

Hf = \
   + 5*i["C-H2C2"]  \
   + 2*i["C-HC3"]   

print(f"\nNorbornane: Hf is {Hf:.2f} kJ/mole.  Strain energy is {-54.9 - Hf:+.2f} kJ/mole")

print(f"Compare to value of +{17.0*4.183:.2f} given in figure 2.15 of book")


Cyclobutanone: Hf is -193.66 kJ/mole.  Strain energy is +102.06 kJ/mole
Cyclobutanol: Hf is -250.34 kJ/mole.  Strain energy is +105.34 kJ/mole
Cyclohexanone: Hf is -276.09 kJ/mole.  Strain energy is +44.99 kJ/mole
Cyclohexanol: Hf is -291.55 kJ/mole.  Strain energy is +0.75 kJ/mole

Methylenecyclobutane: Hf is 8.99 kJ/mole.  Strain energy is +112.51 kJ/mole
Compare to value of 121.31 given in figure 2.15 of book

Norbornane: Hf is -118.92 kJ/mole.  Strain energy is +64.02 kJ/mole
Compare to value of +71.11 given in figure 2.15 of book


### Discussion 
The strain energy does not significantly change between cyclobutanone and cyclobutanol. This is reflected in the heats of hydrogenation of the pair. they are similar. The heat of hydrogenation is for the group alone since strain energy provides no change.

The strain energy does change for cyclohexanone and cyclohexanol.  Cyclohexanone has a strain energy component whereas cyclohexanol has almost none (as expected). We observe an extra 10 or 20 kJ/mole released with hydrogenation of cyclohexanone compared to a linear ketone (no data was found for 2- or 3-hexanone so I used data for 2- and 3-pentanone.)

So cyclohexanol is easier to hydrogenate compared to cyclobutanol. Was that the way you expected it to go?

### The Benefits of a Documented Notebook

Initially my results made no sense. Eventually I realized that I had failed to include the group increment for the alcohol oxygen (i\["O-CH"\]). Having a set of code that I could check helped locate that error. Also, it would have helped others find my mistake if I had not corrected it. Everyone can see exactly how I calculated the heats of formation.

It will also be easy to alter the code for cyclopentanone/cyclopentanol and other pairings. Give that a try.

## References
Below are some places that I found data used in this exercise. If you find useful tables or databases please share with the class.
<hr>

Heats of combustion from CRC handbook: 
https://hbcp-chemnetbase-com.proxy.library.upei.ca/faces/documents/05_05/05_05_0001.xhtml

Heats of Formation from Argonne National laboratory: 
https://atct.anl.gov/Thermochemical%20Data/version%201.122r/index.php

NIST Webbook: 
https://webbook.nist.gov/chemistry/

See also : "Chemical calculations ; an introduction to the use of mathematics in chemistry", Sidney W. Benson, *Wiley \& Sons*, **1971**, UPEI library Stacks (2nd floor) QD42.B453 1971

and

"Thermochemical Data of Organic Compounds", J.B. Pedley, R.D. Naylor, S.P. Kirby, *Chapman and Hall, New York*, **1986**, 1-792 (pirated from the web - honestly, the only legitimate copy that I could find within a library system was in Australia). - There was one error/omission in table 2.4 that I corrected using data from this book.

"Organic Compounds – Physical and Thermochemical Data".  J.G. Speight, *Rules of Thumb for Petroleum Engineers, John Wiley & Sons*, **2017**. 539-545.  https://ebookcentral.proquest.com/lib/upei/detail.action?docID=4812517


In [15]:
# Hf of benzene 

group = i["Cb-H"]
energy = 6 * group

print(f"The energy is {energy:0.2f}")



The energy is 82.76
